lib/client/redirect.go

```diff
@@ -42,6 +42,10 @@ const (
 	// LoginFailedBadCallbackRedirectURL is a redirect URL when an SSO error specific to
 	// auth connector's callback was encountered.
 	LoginFailedBadCallbackRedirectURL = "/web/msg/error/login/callback"
+
+	// LoginFailedUnauthorizedRedirectURL is a redirect URL for when an SSO authenticates successfully,
+	// but the user has no matching roles in Teleport.
+	LoginFailedUnauthorizedRedirectURL = "/web/msg/error/login/auth"
```

Comment:
Maybe:

```suggestion
	LoginFailedUnauthorizedRedirectURL = "/web/msg/error/login/noroles"
```

Reply:
In the frontend, the new error message is `You are not authorized, please contact your SSO administrator.` I figured that there might be other situations where we want to show this message, and tying the URL to roles would be awkward.

Reply:
Fair enough.